runtime/include/RuntimeCAPI.h

```diff
@@ -72,15 +71,44 @@ void __quantum__qis__CRX(double, QUBIT *, QUBIT *);
 void __quantum__qis__CRY(double, QUBIT *, QUBIT *);
 void __quantum__qis__CRZ(double, QUBIT *, QUBIT *);
 void __quantum__qis__CRot(double, double, double, QUBIT *, QUBIT *);
-
 void __quantum__qis__CSWAP(QUBIT *, QUBIT *, QUBIT *);
 void __quantum__qis__Toffoli(QUBIT *, QUBIT *, QUBIT *);
-
 void __quantum__qis__MultiRZ(double, int64_t, /*qubits*/...);
 
+// Adjoint version of quantum gate instructions
+void __quantum__qis__Identity_Adjoint(QUBIT *);
+void __quantum__qis__PauliX_Adjoint(QUBIT *);
+void __quantum__qis__PauliY_Adjoint(QUBIT *);
+void __quantum__qis__PauliZ_Adjoint(QUBIT *);
+void __quantum__qis__Hadamard_Adjoint(QUBIT *);
+void __quantum__qis__S_Adjoint(QUBIT *);
+void __quantum__qis__T_Adjoint(QUBIT *);
+void __quantum__qis__PhaseShift_Adjoint(double, QUBIT *);
+void __quantum__qis__RX_Adjoint(double, QUBIT *);
+void __quantum__qis__RY_Adjoint(double, QUBIT *);
+void __quantum__qis__RZ_Adjoint(double, QUBIT *);
+void __quantum__qis__Rot_Adjoint(double, double, double, QUBIT *);
+void __quantum__qis__CNOT_Adjoint(QUBIT *, QUBIT *);
+void __quantum__qis__CY_Adjoint(QUBIT *, QUBIT *);
+void __quantum__qis__CZ_Adjoint(QUBIT *, QUBIT *);
+void __quantum__qis__SWAP_Adjoint(QUBIT *, QUBIT *);
+void __quantum__qis__IsingXX_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__IsingYY_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__IsingXY_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__IsingZZ_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__ControlledPhaseShift_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__CRX_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__CRY_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__CRZ_Adjoint(double, QUBIT *, QUBIT *);
+void __quantum__qis__CRot_Adjoint(double, double, double, QUBIT *, QUBIT *);
+void __quantum__qis__CSWAP_Adjoint(QUBIT *, QUBIT *, QUBIT *);
+void __quantum__qis__Toffoli_Adjoint(QUBIT *, QUBIT *, QUBIT *);
```

Comment:
Instead of adding this expanded list of new instructions, you could update the signature of original ones to get booleans (or `uint8_t`). For example, the signature of `__quantum__qis__RY` will be updated as follows,

```
void __quantum__qis__RY(double, QUBIT *, uint8_t);
```

Check the abstract method of NamedOperation in [QuantumDevice](https://github.com/PennyLaneAI/catalyst/blob/b5df1720cf07d78336de09f729cbd2e688406f90/runtime/include/QuantumDevice.hpp#L134). You can control inverse (adjoint) with simply passing true/false.

Reply:
Right, the current approach is more in line with QIR conventions, while what you propose is a smaller API.
I think for this PR I don't think it matters which way we go, because maybe we will decide to more strictly adhere to QIR in the future or maybe we go the opposite way, which is hard to say right now.
So from my side you can merge it in which ever way (since we might revamp it in the near future anyway).

@maliasadi We might want to set aside some time to brainstorm about the C-API and what it should look like.

Reply:
@maliasadi the latest version should use `adjoint` parameter rather than a suffixed functions.